### PR TITLE
chore: api test

### PR DESCRIPTION
**Pull request by chandumlg**
## What are the changes introduced in this PR?

Adds
- Service api tests
- Controller tests
- Service tests

## What is the related Linear task?

Resolves INT-1350
Resolves INT-1417
- https://linear.app/rudderstack/issue/INT-1350/api-tests-with-a-test-destination
- https://linear.app/rudderstack/issue/INT-1417/add-service-and-controller-tests

## Please explain the objectives of your changes below

Put down any required details on the broader aspect of your changes. If there are any dependent changes, **mandatorily** mention them here

### Any changes to existing capabilities/behaviour, mention the reason & what are the changes ?

N/A

### Any new dependencies introduced with this change?

N/A

### Any new generic utility introduced or modified. Please explain the changes.

N/A

### Any technical or performance related pointers to consider with the change?

N/A

<hr>

### Developer checklist

- [ ] My code follows the style guidelines of this project

- [ ] **No breaking changes are being introduced.**

- [ ] All related docs linked with the PR?

- [ ] All changes manually tested?

- [ ] Any documentation changes needed with this change?

- [ ] Is the PR limited to 10 file changes?

- [ ] Is the PR limited to one linear task?

- [ ] Are relevant unit and component test-cases added?

### Reviewer checklist

- [ ] Is the type of change in the PR title appropriate as per the changes?

- [ ] Verified that there are no credentials or confidential data exposed with the changes.

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Review by koladilip — DISMISSED** [review]

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/2995/test-report.html

**Review by koladilip — DISMISSED** [review]

File: src/controllers/regulation.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ export class RegulationController {
         rudderDestInfo,
       );
       ctx.body = resplist;
-      ctx.status = resplist[0].statusCode;
+      ctx.status = resplist[0].statusCode; // TODO: check if this is the right way to set status
```
Comment (chandumlg) [inline src/controllers/regulation.ts:37]: We might need to address them some time in the future @krishna2020

**Review by utsabc — DISMISSED** [review]

**Comment by utsabc** [comment]
Couple of points we need to take care
- The test data for now is a simplified version of exact data we would get from server when we incorporate zod these we need replicate exact prod structure (this only affects the test data) can be taken in future as we integrate zod
- Currently we are focussing on native integration service as it is the major service in future we need to expact to other services like cdkv2, plugin as we onboard more destinations to them
- We need to discuss the current limitations we are having with regulation service with server team, also there are some issues in the code we need to address

**Review by utsabc — APPROVED** [review]

**Review by sanpj2292 — APPROVED** [review]

**Pull request merged**